generator/internal/metadata_decorator_generator.cc

```diff
@@ -406,6 +406,11 @@ void $metadata_class_name$::SetMetadata(grpc::ClientContext& context,
   }
   auto const& authority = options.get<AuthorityOption>();
   if (!authority.empty()) context.set_authority(authority);
+  if (options.has<google::cloud::CustomHeadersOption>()) {
```

Comment:
Optional nit:  It looks like we're a little inconsistent with our qualifying.  `Options` a few lines up, for example, is mentioned without `google::cloud::`.